src/useErrorBoundary.ts

```diff
@@ -2,15 +2,25 @@ import { useContext, useMemo, useState } from "react";
 import { assertErrorBoundaryContext } from "./assertErrorBoundaryContext";
 import { ErrorBoundaryContext } from "./ErrorBoundaryContext";
 
+type StateError<TError> =
+  | (TError & { isShowBoundary: true })
+  | (Error & { isShowBoundary: true });
```

Comment:
I think React _used to use_ `_suppressLogging` internally for a similar purpose (see https://github.com/facebook/react/pull/13384)

I like that name a bit better than "isShowBoundary" because it describes the desired _behavior_.

That being said, this behavior was recently removed (https://github.com/facebook/react/pull/28515) so I guess we can't just use that attribute to suppress React's own error logging. Another note on that PR that seems relevant here (and RE: my comment above about the event handler):
> We broke the ability to "break on uncaught exceptions" by adding a try/catch higher up in the scheduling. 

We should be careful _not_ to break this.

Reply:
Change `isShowBoundary` to `_suppressLogging`. In the process of reworking and testing, I also noticed that when `error` has `_suppressLogging` as a property, the `console.error` function would not be called, therefore there is no need to override `console.error`, at least for the time being.